### PR TITLE
style: Refactor some animations code

### PR DESCRIPTION
**Pull request by mrobinson**
This change modifies the names of some methods to make it clearer what
they are doing. It also adds some clarifying comments to explain some
confusing behavior.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because they do not change behavior.

### CODE REVIEW

File: components/style/matching.rs [hunk 1]
```diff
@@ -588,46 +603,50 @@ trait PrivateMatchMethods: TElement {
...
-    //
-    // Also, this function gets the old style, for some reason I don't really
-    // get, but the functions called (mainly update_style_for_animation) expects
-    // the new style, wtf?
     #[cfg(feature = "servo")]
-    fn update_animations_for_cascade(
+    fn collect_and_update_style_for_expired_transitions(
         &self,
         context: &SharedStyleContext,
+        node: OpaqueNode,
```
Comment (emilio) [inline components/style/matching.rs:610]: `node` is always `self.as_node().opaque()`. Either assert it, or make this a static function as it doesn't use `self`?

File: components/style/matching.rs [hunk 2]
```diff
@@ -588,46 +603,50 @@ trait PrivateMatchMethods: TElement {
...
+                    frame.property_animation.update(Arc::make_mut(style), 1.0);
+                    expired_transitions.push(frame.property_animation);
+                }
+            }
+        }
+    }
+
+    #[cfg(feature = "servo")]
+    fn update_style_for_animations_and_collect_running_transitions(
+        &self,
```
Comment (emilio) [inline components/style/matching.rs:632]: Same, this no longer uses `self`, could be made a standalone function.

**Review by emilio — APPROVED** [review]
r=me with those two cleanups.

File: components/style/matching.rs (hunk 1 above)
Comment (mrobinson) [reply components/style/matching.rs:609]: Thanks for the review! I've converted these two to static methods, just for code organization purposes.

**Comment by mrobinson** [comment]
@bors-servo retry
  * #24622
  * #25513

**Pull request merged**